src/main/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo.java

```diff
@@ -605,70 +610,118 @@ private void checkDeprecatedParameterUsage( Object parameter, String name, Strin
         }
     }
 
-    private int countViolations( XmlPullParser xpp )
+    private List<Violation> getViolations( XmlPullParser xpp )
         throws XmlPullParserException, IOException
     {
-        int count = 0;
-        int ignoreCount = 0;
-        List<RuleUtil.Matcher> ignores = violationIgnore == null ? Collections.<RuleUtil.Matcher>emptyList()
-                        : RuleUtil.parseMatchers( violationIgnore.split( "," ) );
+        List<Violation> violations = new ArrayList<>();
 
         String basedir = project.getBasedir().getAbsolutePath();
         String file = "";
+
         for ( int eventType = xpp.getEventType(); eventType != XmlPullParser.END_DOCUMENT; eventType = xpp.next() )
         {
             if ( eventType != XmlPullParser.START_TAG )
             {
                 continue;
             }
-            else if ( "file".equals( xpp.getName() ) )
+
+            if ( "file".equals( xpp.getName() ) )
```

Comment:
The else ifs were clearer, even if not strictly necessary. A single if block without continues might be clearer still.

Reply:
Well if you really wanted to, I can rework this. I do only see disadvantages and probably most ppl will read `if (condition) continue;` as a guard statement:

https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Please also note that else if will make refactoring more complex, i.e. if adding code. Without the ifs, we could run into two `if`blocks if we wanted to. Using only  `if … else if`, we do not have this option and need to refactor for guard statements anyway.

The docs did not say anything about "do not use guard statements". I can put them back in, if you like.

Reply:
The point is the else if makes it clear that we don't expect multiple paths to be taken. As to how someone might need to change (not refactor) this block in the future, there's no way to predict that. Just make this as clear as it can be for what we need now.

Reply:
I also prefer the 'else if' approach
+1 to @elharo suggestion